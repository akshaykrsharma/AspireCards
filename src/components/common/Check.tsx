import React, {useState, useEffect} from 'react';
import {Text, View, Image, TouchableOpacity} from 'react-native';
import {CheckedPropType} from '../../interfaces/interface';
import Images from '../../res/Images';

function Check(props: CheckedPropType) {
  const [checked, setChecked] = useState(props.isChecked);
  // useEffect(() => {
  // props.selectedValue &&  props.selectedValue(checked);
  // }, [checked]);

  return (
    <View style={{flexDirection: 'row', alignItems: 'center'}}>
      <TouchableOpacity
        onPress={() => {
          props.selectedValue && props.selectedValue(!checked);
          setChecked(!checked);
        }}>
        <Image
          style={Images.checked.style}
          source={checked ? Images.checked.source : Images.unChecked.source}
        />
      </TouchableOpacity>
      {props.label && <Text style={{marginLeft: 5}}>{props.label}</Text>}
    </View>
  );
}

export default Check;
