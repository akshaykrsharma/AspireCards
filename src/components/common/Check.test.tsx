import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer'

import Check from './Check';

//it is alias of test so we can use both

describe("Checking Check Component...", () => {
	it('Check has been rendered successfully', () => {
		renderer.create(<Check isChecked />);
	});
});


/// To creating snapshot

describe("Checking for snapshot...", () => {
	test("Taking snapshot of code", () => {
		const component = renderer.create(<Check isChecked />);
		const contentOfCheck = component.toJSON();
		expect(contentOfCheck).toMatchSnapshot();
	});
});




