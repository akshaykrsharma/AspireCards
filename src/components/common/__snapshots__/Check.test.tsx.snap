// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checking for snapshot... Taking snapshot of code 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "row",
    }
  }
>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <Image
      source={
        Object {
          "testUri": "../../../src/res/img/checked.png",
        }
      }
      style={Object {}}
    />
  </View>
</View>
`;
